pilot/pkg/model/xds_cache.go

```diff
@@ -121,6 +156,77 @@ func (c *inMemoryCache) Keys() []string {
 	return keys
 }
 
+type lruCache struct {
+	mu          sync.RWMutex
+	store       *lru.Cache
+	configIndex map[ConfigKey]sets.Set
+}
+
+var _ XdsCache = &lruCache{}
+
+func newLru() *lru.Cache {
+	l, err := lru.New(defaultLRUSize)
+	if err != nil {
+		panic(fmt.Errorf("invalid lru configuration: %v", err))
+	}
+	return l
+}
+
+func (l *lruCache) Add(entry XdsCacheEntry, value *any.Any) {
+	if !entry.Cacheable() {
+		return
+	}
+	l.mu.Lock()
+	defer l.mu.Unlock()
+	k := entry.Key()
+	l.store.Add(k, value)
+	indexConfig(l.configIndex, entry.Key(), entry)
+	writes.Increment()
+}
+
+func (l *lruCache) Get(entry XdsCacheEntry) (*any.Any, bool) {
+	if !entry.Cacheable() {
+		return nil, false
+	}
+	l.mu.RLock()
```

Comment:
We do not need this lock. lru cache already has an internal lock

Reply:
ClearAll could remove the member `store` from `l`.

Reply:
I see. would it help if we use `Purge` for ClearAll ?